### PR TITLE
feat: Implement Developer Mode

### LINKED ISSUE #1739: Implement Developer Mode by default and hide "Tickets" and "Edit Events" if Dev Mode is switched off
In the device settings add a mode "Developer Mode". 
1. This mode should be deactivated by default. 
2. Then in the event menus hide the following. Only show this if users switch on "Developer Mode":
   * "Tickets"
   * "Edit Event"
   * "+" icon to create new events.
   * "Orders" 
-> Only show these features in "Developer Mode!
3. When people activate developer mode, please show a pop up "You are activating Developer Mode. This mode will give you access to unreleased app features. These features are unstable and could potentially result in unwanted changes in your event or data loss. Use with care! Do you really want to activate developer mode? [Yes, I am taking my Chances] [No, I want to stay safe]
4. Implement the "Yes" button in red.

### PR DESCRIPTION
**Pull request by ShridharGoel**
Fixes #1739 

Changes: 

Added Developer Mode in Device Settings. 
Enabling the Developer Mode provides access to hidden features of the app.

GIF for the change:

ezgif com-video-to-gif (91)

### CODE REVIEW

File: app/src/main/java/com/eventyay/organizer/core/main/MainActivity.java [hunk 1]
```diff
@@ -141,7 +146,17 @@ public void onDrawerClosed(View drawerView) {
     public void setEventId(long eventId) {
         this.eventId = eventId;
         fragmentNavigator.setEventId(eventId);
-        binding.navView.getMenu().setGroupVisible(R.id.subMenu, true);
+        binding.navView.getMenu().findItem(R.id.nav_dashboard).setVisible(true);
+        binding.navView.getMenu().findItem(R.id.nav_attendees).setVisible(true);
+        binding.navView.getMenu().findItem(R.id.nav_share).setVisible(true);
+        binding.navView.getMenu().findItem(R.id.nav_about_event).setVisible(true);
+        binding.navView.getMenu().findItem(R.id.nav_event_settings).setVisible(true);
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/main/MainActivity.java]: Instead, create a list of normal features and iterate

File: app/src/main/java/com/eventyay/organizer/utils/Utils.java [hunk 2]
```diff
@@ -105,4 +107,8 @@ public static Bitmap drawableToBitmap (Drawable drawable) {
 
         return bitmap;
     }
+
+    public static void toggleDeveloperMode() {
+        isDeveloperModeEnabled = !isDeveloperModeEnabled;
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/utils/Utils.java]: Should be in shared preference

Comment (ShridharGoel) [reply app/src/main/java/com/eventyay/organizer/utils/Utils.java]: Updated.

File: app/src/main/java/com/eventyay/organizer/core/main/MainActivity.java (hunk 1 above)
Comment (ShridharGoel) [reply app/src/main/java/com/eventyay/organizer/core/main/MainActivity.java]: Done.

**Comment by iamareebjamal** [comment]
PRF

**Comment by mariobehling** [comment]
I don't see a dialogue when switching off Developer Mode. Please also implement that.

File: app/src/main/java/com/eventyay/organizer/core/main/MainActivity.java [hunk 3]
```diff
@@ -141,10 +160,22 @@ public void onDrawerClosed(View drawerView) {
...
+
+        if (isDeveloperModeEnabled) {
+            for (Integer itemId : drawerExtraItems) {
+                binding.navView.getMenu().findItem(itemId).setVisible(true);
+            }
+        }
 
         if (BuildConfig.HIDE_DRAWER_ITEMS) {
-            for (Integer itemId : drawerItems) {
+            for (Integer itemId : drawerExtraItems) {
```
Comment (anhanh11001) [inline app/src/main/java/com/eventyay/organizer/core/main/MainActivity.java]: I don't get why we have `BuildConfig.HIDE_DRAWER_ITEMS` and 
`sharedPreferences.getBoolean( getString(R.string.developer_mode_key), false);`. Should one of them be removed?

File: app/src/main/java/com/eventyay/organizer/core/main/MainActivity.java [hunk 4]
```diff
@@ -39,15 +43,26 @@
...
+        R.id.nav_attendees,
+        R.id.nav_share,
+        R.id.nav_about_event,
+        R.id.nav_event_settings);
+
+    private final List<Integer> drawerExtraItems = Arrays.asList(
+        R.id.nav_sell,
+        R.id.nav_orders,
+        R.id.nav_tickets,
+        R.id.nav_edit_event,
```
Comment (anhanh11001) [inline app/src/main/java/com/eventyay/organizer/core/main/MainActivity.java]: Why edit event is not in the main drawer list? Is it what we want?

File: app/src/main/java/com/eventyay/organizer/core/settings/SettingsFragment.java [hunk 5]
```diff
@@ -94,6 +100,51 @@ public void onCreatePreferencesFix(@Nullable Bundle bundle, String rootKey) {
...
+
+        findPreference("developer_mode").setOnPreferenceClickListener(preference -> {
+            isDeveloperModeEnabled = manager.getSharedPreferences().getBoolean(
+                getString(R.string.developer_mode_key), false);
+
+            if (!isDeveloperModeEnabled) {
+                AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
+                builder.setTitle(R.string.activate_developer_mode);
+                builder.setMessage(R.string.developer_mode_activation_message);
+                builder.setPositiveButton("Yes, I am taking my chances",
```
Comment (anhanh11001) [inline app/src/main/java/com/eventyay/organizer/core/settings/SettingsFragment.java]: Extract string here

File: app/src/main/java/com/eventyay/organizer/core/settings/SettingsFragment.java [hunk 6]
```diff
@@ -94,6 +100,51 @@ public void onCreatePreferencesFix(@Nullable Bundle bundle, String rootKey) {
...
+                    (dialog, which) -> {
+                        toggleDeveloperMode();
+                        Intent intent = new Intent(getActivity(), MainActivity.class);
+                        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+                        startActivity(intent);
+                    });
+                builder.setNegativeButton("No, I want to stay safe",
```
Comment (anhanh11001) [inline app/src/main/java/com/eventyay/organizer/core/settings/SettingsFragment.java]: Here as well

File: app/src/main/java/com/eventyay/organizer/core/settings/SettingsFragment.java [hunk 7]
```diff
@@ -94,6 +100,51 @@ public void onCreatePreferencesFix(@Nullable Bundle bundle, String rootKey) {
...
+
+                dialog.getButton(DialogInterface.BUTTON_POSITIVE)
+                    .setTextColor(getResources().getColor(R.color.red_500));
+                dialog.getButton(DialogInterface.BUTTON_NEGATIVE)
+                    .setTextColor(getResources().getColor(R.color.green_500));
+            } else {
+                AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
+                builder.setTitle(R.string.deactivate_developer_mode);
+                builder.setMessage(R.string.developer_mode_deactivation_message);
+                builder.setPositiveButton("Yes",
```
Comment (anhanh11001) [inline app/src/main/java/com/eventyay/organizer/core/settings/SettingsFragment.java]: Here

File: app/src/main/java/com/eventyay/organizer/core/settings/SettingsFragment.java [hunk 8]
```diff
@@ -94,6 +100,51 @@ public void onCreatePreferencesFix(@Nullable Bundle bundle, String rootKey) {
...
+                    (dialog, which) -> {
+                        toggleDeveloperMode();
+                        Intent intent = new Intent(getActivity(), MainActivity.class);
+                        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+                        startActivity(intent);
+                    });
+                builder.setNegativeButton("No",
```
Comment (anhanh11001) [inline app/src/main/java/com/eventyay/organizer/core/settings/SettingsFragment.java]: Here

File: app/src/main/java/com/eventyay/organizer/core/main/MainActivity.java (hunk 4 above)
Comment (ShridharGoel) [reply app/src/main/java/com/eventyay/organizer/core/main/MainActivity.java]: Yes it has to be hidden for now.

File: app/src/main/java/com/eventyay/organizer/core/main/MainActivity.java (hunk 3 above)
Comment (ShridharGoel) [reply app/src/main/java/com/eventyay/organizer/core/main/MainActivity.java]: I'm not sure about the use of `HIDE_DRAWER_ITEMS`. It is set to true for the release version and false for the debug version. 
@iamareebjamal Should it be removed?

File: app/src/main/java/com/eventyay/organizer/core/settings/SettingsFragment.java (hunk 5 above)
Comment (ShridharGoel) [reply app/src/main/java/com/eventyay/organizer/core/settings/SettingsFragment.java]: Updated.

File: app/src/main/java/com/eventyay/organizer/core/settings/SettingsFragment.java (hunk 7 above)
Comment (ShridharGoel) [reply app/src/main/java/com/eventyay/organizer/core/settings/SettingsFragment.java]: Updated.

**Comment by ShridharGoel** [comment]
> I don't see a dialogue when switching off Developer Mode. Please also implement that.

@mariobehling Done.

File: app/src/main/java/com/eventyay/organizer/core/main/MainActivity.java (hunk 3 above)
Comment (iamareebjamal) [reply app/src/main/java/com/eventyay/organizer/core/main/MainActivity.java]: Yes, remove it

**Comment by liveHarshit** [comment]
There is no icon for roles in the drawer menu.

**Pull request merged**